fairlearn/metrics/_metric_frame.py

```diff
@@ -119,24 +133,162 @@ def __init__(self,
                  y_pred, *,
```

Comment:
As a warning #822 is changing this signature slightly. There may be merge conflicts.